parser/ParseIndexProperties.cpp

```diff
@@ -0,0 +1,28 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "parser/ParseIndexProperties.hpp"
+
+#include <string>
+
+namespace quickstep {
+  // initialize constants for various index properties
+  const std::string BloomFilterIndexProperties::kBloomFilterSizeInBytes = "size";
```

Comment:
@zuyu I see that you may be referring to the use of the `hasInvalidCustomProperty() `and `hasDuplicateCustomProperty()` by only the `BloomFilterIndexProperties` derived class. 
I intended them to be more of like helper routines that will be invoked by all the derived classes to check for duplicates and invalid properties. Otherwise this piece of code may be duplicated by all the other classes. 
Currently, CSBTreeIndex does not define much of additional tunable properties (at least that I know of),  but a BitWeavingIndex, or SMAIndex might define some. I thought this might come handy during that time. What do you suggest?